src/main/resources/com/google/api/codegen/java/settings.snip

```diff
@@ -147,18 +146,20 @@
 
   private static String getGapicVersion() {
     if (gapicVersion == null) {
-      try {
-        Properties properties = new Properties();
-        String mavenPropertiesPath =
-            META_FILE_ROOT + GROUP_ID + "/" + ARTIFACT_ID + "/pom.properties";
-        InputStream inputStream =
-            ImageAnnotatorSettings.class.getResourceAsStream(mavenPropertiesPath);
-        if (inputStream != null) {
-          properties.load(inputStream);
-          gapicVersion = properties.getProperty(META_VERSION_KEY, DEFAULT_GAPIC_VERSION);
-        }
-      } catch (Exception ignore) {
+      gapicVersion = {@xsettingsClass.name}.class.getPackage().getImplementationVersion();
+      if (gapicVersion == null) {
         gapicVersion = DEFAULT_GAPIC_VERSION;
+        try {
+          Properties properties = new Properties();
+          InputStream inputStream =
+              {@xsettingsClass.name}.class.getResourceAsStream(PROPERTIES_FILE);
+          if (inputStream != null) {
+            properties.load(inputStream);
+            gapicVersion = properties.getProperty(META_VERSION_KEY, DEFAULT_GAPIC_VERSION);
+          }
+        } catch (Exception ignore) {
+          // Ignore
+        }
```

Comment:
Could we maybe make a gax helper function so that this same code doesn't have to be generated for every API?